Source/Artifacts/TypeCannotHaveMultipleArtifactAssociations.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Dolittle.SDK.Artifacts
+{
+    /// <summary>
+    /// Exception that gets thrown when a <see cref="Type" /> is associated with multiple <see cref="Artifact" />.
+    /// </summary>
+    public class TypeCannotHaveMultipleArtifactAssociations : Exception
```

Comment:
Should there also be a converse error - so an Artifact cannot be associated to multiple types?
It seems like that is just ignored right now.

Reply:
It is ignored, also ignored in JS SDK, We should track it IMO and throw an exception.

Should we check for multiple associations in the Artifacts implementaiton as well?